### PR TITLE
Vector of shared ptrs homework

### PR DESCRIPTION
**Pull request by raxonpc**
@raxonpc, @ploWoj

### CODE REVIEW

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @raxonpc

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @ploWoj

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @raxonpc

**Comment by ziobron** [comment]
⏱ 3 XP extra for @raxonpc for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @ploWoj

**Comment by ziobron** [comment]
⏱ 3 XP extra for @ploWoj for on-time delivery

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        if(!(std::any_of(pass1.begin(),pass1.end(), ::isdigit))){
+            return ErrorCode::PasswordNeedsAtLeastOneNumber;
+        }
+        if(!(std::any_of(pass1.begin(),pass1.end(), ::ispunct))){
+            return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+        }
+        if(!(std::any_of(pass1.begin(),pass1.end(), ::isupper))){
+            return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
+        }
+
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:48]: Add space after `if`, `for`, `while` `function` etc...

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+bool doPasswordsMatch(const std::string& pass1, const std::string& pass2)
+{
+    return pass1 == pass2;
+}
+
+ErrorCode checkPasswordRules(const std::string& pass1)
+    {
+        if(pass1.length() < 9)
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:34]: Add space after `if`, `for`, `while` `function` etc...

File: homework/password-check/validation.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            return ErrorCode::PasswordNeedsAtLeastNineCharacters;
+        }
+
+        if(!(std::any_of(pass1.begin(),pass1.end(), ::isdigit))){
+            return ErrorCode::PasswordNeedsAtLeastOneNumber;
+        }
+        if(!(std::any_of(pass1.begin(),pass1.end(), ::ispunct))){
+            return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+        }
+        if(!(std::any_of(pass1.begin(),pass1.end(), ::isupper))){
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:45]: better use none_of

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**